### PR TITLE
Fail for llvm parsing errors in diff-diff.rb

### PR DESCRIPTION
**Pull request by elliottt**
When llvm-diff fails because of parsing errors, diff-diff.rb should fail as well. This PR fixes this bug in diff-diff.rb and adds a disabled failing test for exp validation to verify that it's working.

### Motivation

Fixing a bug in validate_exp.sh that was not reporting llvm parse errors as test failures.

### Test plan

See included automated tests.

### CODE REVIEW

File: test/testdata/compiler/disabled/validate_exp_tests/test.llo.exp [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  unreachable
+}
+
+; Function Attrs: sspreq
+define void @Init_test() local_unnamed_addr #3 {
+entry:
+  %locals.i.i = alloca i64, i32 0, align 8
+
+  ; Call a function that doesn't exist to force a error that won't show up as a diff
+  %realpath = tail call i64 @sorbet_readRealpath_fake()
```
Comment (aprocter-stripe) [inline test/testdata/compiler/disabled/validate_exp_tests/test.llo.exp:140]: So it looks like we have manually injected an error here. Will `update_compiler_exp.sh` clobber this if run?

File: test/validate_exp.sh [hunk 2]
```diff
@@ -81,14 +101,23 @@ for ext in "${exts[@]}"; do
         if grep "exists only in" "$diff_out" > /dev/null ; then
           cat "$diff_out"
           info "If this was an expected difference, you need to run tools/scripts/update_compiler_exp.sh"
-          fatal "* $(basename "$exp")"
+          something_failed
```
Comment (aprocter-stripe) [inline test/validate_exp.sh:104]: Might be nice to pass `"$exp"` into `something_failed` and inject it into the error/success message there.

**Review by aprocter-stripe — APPROVED** [review]
Looks good to me, a couple questions you might want to consider before merge.

File: test/validate_exp.sh (hunk 2 above)
Comment (elliottt) [reply test/validate_exp.sh:104]: That seems totally reasonable 👍

File: test/testdata/compiler/disabled/validate_exp_tests/test.llo.exp (hunk 1 above)
Comment (elliottt) [reply test/testdata/compiler/disabled/validate_exp_tests/test.llo.exp:140]: That's a really good point, I wonder if the best way to do this is to just exempt the `disabled/validate_exp_tests` path from exp updates?

Comment (aprocter-stripe) [reply test/testdata/compiler/disabled/validate_exp_tests/test.llo.exp:140]: I think that probably makes sense! Racking my brain a bit, I can't imagine the use of updating `exp` files for a disabled test.

Comment (elliottt) [reply test/testdata/compiler/disabled/validate_exp_tests/test.llo.exp:140]: I took that approach for now. If there's a better way to maintain the hand-edits, we can make that change in the future :)

**Pull request merged**